library/src/main/java/org/apache/apex/malhar/lib/utils/serde/AffixedBuffer.java

```diff
@@ -0,0 +1,112 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.apex.malhar.lib.utils.serde;
+
+/**
+ * Unlike LengthValueBuffer which start with the length,
+ * This serialize buffer prefix/suffix with a fixed byte array.
+ * This SerializeBuffer can be used for key which prefixed/suffixed by identifier
+ *
+ */
+public class AffixedBuffer extends AbstractSerializeBuffer
+{
+  protected byte[] prefix;
+  protected byte[] suffix;
+
+  /**
+   * whether serialize the length of value. It is not necessary when serialize the key.
+   * But the previous implementation keep the length. Default is true to compatible with previous implementation.
```

Comment:
I don't think we have to worry about being compatible with previous implementation since spillable DS is not used widely yet